common/src/main/java/org/apache/falcon/entity/ProcessHelper.java

```diff
@@ -185,4 +187,36 @@ private static void validateProcessInstance(Process process, Date instanceTime,
         }
         return result;
     }
+
+    public static Validity getClusterValidity(Process process, String clusterName) throws FalconException {
+        org.apache.falcon.entity.v0.process.Cluster cluster = getCluster(process, clusterName);
+        if (cluster == null) {
+            throw new FalconException("Invalid cluster: " + clusterName + " for process: " + process.getName());
+        }
+        return cluster.getValidity();
+    }
+
+    public static Sla getSLA(String clusterName, Process process) {
+        Cluster cluster = getCluster(process, clusterName);
+        return cluster != null ? getSLA(cluster, process) : null;
+    }
+
+    public static Sla getSLA(Cluster cluster, Process process) {
+        final Sla clusterSla = cluster.getSla();
+        if (clusterSla != null) {
+            return clusterSla;
+        }
+        final Sla processSla = process.getSla();
+        return processSla == null ? null : processSla;
+    }
+
+    public static Date getProcessValidityStart(Process process, String clusterName) throws FalconException {
```

Comment:
in this method we just checking if the cluster that we have send to the function exits in the process definition...if we remove the method we have to do that check there.
Let me know your thought ? We have same kind of implementation for feed also.